plugins/module_utils/network/sonic/config/bgp_af/bgp_af.py

```diff
@@ -531,50 +583,85 @@ def get_delete_route_advertise_route_map_request(self, vrf_name, conf_afi, conf_
-                                            conf_safi = conf_addr_fam.get('safi', None)
-                                            conf_route_advertise_list = conf_addr_fam.get('route_advertise_list', [])
-                                            if conf_route_advertise_list:
-                                                for conf_rt_adv in conf_route_advertise_list:
-                                                    conf_advertise_afi = conf_rt_adv.get('advertise_afi', None)
-                                                    conf_route_map = conf_rt_adv.get('route_map', None)
-                                                    # Deletion at route-advertise level
-                                                    if (not advertise_afi and vrf_name == conf_vrf_name and afi == conf_afi and safi == conf_safi):
-                                                        requests.append(self.get_delete_route_advertise_request(vrf_name, afi, safi))
-                                                    # Deletion at advertise-afi-safi level
-                                                    if (advertise_afi and not route_map and vrf_name == conf_vrf_name and afi == conf_afi and safi ==
-                                                            conf_safi and advertise_afi == conf_advertise_afi):
-                                                        requests.append(self.get_delete_route_advertise_list_request(vrf_name, afi, safi, advertise_afi))
-                                                    # Deletion at route-map level
-                                                    if (route_map and vrf_name == conf_vrf_name and afi == conf_afi and safi == conf_safi
-                                                            and advertise_afi == conf_advertise_afi and route_map == conf_route_map):
-                                                        requests.append(self.get_delete_route_advertise_route_map_request(vrf_name, afi, safi,
-                                                                                                                          advertise_afi, route_map))
+        if is_delete_all:
+            requests.append(self.get_delete_route_advertise_request(vrf_name, conf_afi, conf_safi))
+        else:
+            for conf_rt_adv in conf_route_adv_list:
+                conf_advertise_afi = conf_rt_adv.get('advertise_afi', None)
+                conf_route_map = conf_rt_adv.get('route_map', None)
+                # Check if the commands to be deleted are configured
+                for mat_rt_adv in mat_route_adv_list:
+                    mat_advertise_afi = mat_rt_adv.get('advertise_afi', None)
+                    mat_route_map = mat_rt_adv.get('route_map', None)
+                    # Deletion at advertise-afi-safi level
+                    if (conf_advertise_afi and not conf_route_map and conf_advertise_afi == mat_advertise_afi):
```

Comment:
Yes, the check is unnecessary so I've removed it. 

I removed the route-advertise deletion because it wasn't actually doing anything. As we discussed previously for other modules, leaf list delete isn't possible without a key. Specifying `route_advertise_list:` in the yml is interpreted as null and did nothing which is why it passed.